dist/smoothscroll.js

```diff
@@ -243,6 +244,43 @@
       );
     };
 
+    // Element.prototype.scroll and Element.prototype.scrollTo
+    Element.prototype.scroll = Element.prototype.scrollTo = function() {
+      // avoid smooth behavior if not required
+      if (shouldBailOut(arguments[0])) {
+        original.elScroll.call(
+            this,
+            arguments[0].left || arguments[0],
+            arguments[0].top || arguments[1]
+        );
+        return;
+      }
+
+      // LET THE SMOOTHNESS BEGIN!
+      smoothScroll.call(
+          this,
+          this,
+          arguments[0].left,
+          arguments[0].top
+      );
+    };
+
+    // Element.prototype.scrollBy
+    Element.prototype.scrollBy = function() {
+      var arg0 = arguments[0];
+
+      if (typeof arg0 === 'object') {
+        arg0.left += this.scrollLeft;
+        arg0.top += this.scrollTop;
```

Comment:
We shouldn’t mutate incoming arguments. We’ll likely want to introduce a really small and simple `assign` method to handle this.

```
this.scroll(assign({}, arg0, {
  left: this.scrollLeft + arg0.left,
  top: this.scrollTop + arg0.top,
  behavior: arg0.behavior,
}));
```